File: app/src/main/java/com/warchaser/musicplayer/tools/MusicInfo.java
package com.warchaser.musicplayer.tools;

import android.net.Uri;
import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by Wu on 2014/10/20.
 *
 */
public class MusicInfo implements Parcelable
{

    private long lId;
    private String sTitle;
    private String sAlbum;
    private long mAlbumId;
    private int iDuration;
    private long lSize;
    private String sArtist;
    private String sUrl;
    private String pinyinInitial;
    private String pinyinTitle;
    private String mUriWithCoverPic;

    //Getters & setters

    public MusicInfo(Parcel in)
    {
        lId = in.readLong();
        sTitle = in.readString();
        sAlbum = in.readString();
        mAlbumId = in.readLong();
        iDuration = in.readInt();
        lSize = in.readLong();
        sArtist = in.readString();
        sUrl = in.readString();
        pinyinInitial = in.readString();
        pinyinTitle = in.readString();
        mUriWithCoverPic = in.readString();

    }

    public static final Creator<MusicInfo> CREATOR = new Creator<MusicInfo>()
    {
        @Override
        public MusicInfo createFromParcel(Parcel in)
        {
            return new MusicInfo(in);
        }

        @Override
        public MusicInfo[] newArray(int size)
        {
            return new MusicInfo[size];
        }
    };

    public long getId() {
        return lId;
    }

    public void setId(long id) {
        this.lId = id;
    }

    public String getTitle() {
        return sTitle;
    }

    public void setTitle(String title) {
        this.sTitle = title;
    }

    public String getAlbum() {
        return sAlbum;
    }

    public void setAlbum(String album) {
        this.sAlbum = album;
    }

    public long getAlbumId()
    {
        return mAlbumId;
    }

    public void setAlbumId(long albumId)
    {
        this.mAlbumId = albumId;
    }

    public int getDuration() {
        return iDuration;
    }

    public void setDuration(int duration) {
        this.iDuration = duration;
    }

    public long getSize() {
        return lSize;
    }

    public void setSize(long size) {
        this.lSize = size;
    }

    public String getArtist() {
        return sArtist;
    }

    public void setArtist(String artist) {
        this.sArtist = artist;
    }

    public String getUrl() {
        return sUrl;
    }

    public void setUrl(String url) {
        this.sUrl = url;
    }

    public String getPinyinInitial() {
        return pinyinInitial;
    }

    public void setPinyinInitial(String pinyinInitial) {
        this.pinyinInitial = pinyinInitial;
    }

    public String getPinyinTitle() {
        return pinyinTitle;
    }

    public void setPinyinTitle(String pinyinTitle) {
        this.pinyinTitle = pinyinTitle;
    }

    public String getUriWithCoverPic()
    {
        return mUriWithCoverPic;
    }

    public void setUriWithCoverPic(String uri)
    {
        this.mUriWithCoverPic = uri;
    }

    public void setUriWithCoverPic(Uri uri)
    {
        this.mUriWithCoverPic = uri.toString();
    }

    public MusicInfo(){

    }

    public MusicInfo(long pId, String pTitle)
    {
        lId = pId;
        sTitle = pTitle;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i)
    {
        parcel.writeLong(lId);
        parcel.writeString(sTitle);
        parcel.writeString(sAlbum);
        parcel.writeLong(mAlbumId);
        parcel.writeInt(iDuration);
        parcel.writeLong(lSize);
        parcel.writeString(sArtist);
        parcel.writeString(sUrl);
        parcel.writeString(mUriWithCoverPic);
    }

    public static final Creator<MusicInfo>
        creator = new Creator<MusicInfo>()
    {
        @Override
        public MusicInfo createFromParcel(Parcel parcel)
        {
            MusicInfo musicInfo = new MusicInfo();
            musicInfo.setId(parcel.readLong());
            musicInfo.setTitle(parcel.readString());
            musicInfo.setAlbum(parcel.readString());
            musicInfo.setAlbumId(parcel.readLong());
            musicInfo.setArtist(parcel.readString());
            musicInfo.setUrl(parcel.readString());
            musicInfo.setDuration(parcel.readInt());
            musicInfo.setSize(parcel.readLong());
            musicInfo.setUriWithCoverPic(parcel.readString());
            return musicInfo;
        }

        @Override
        public MusicInfo[] newArray(int i)
        {
            return new MusicInfo[i];
        }
    };
}

File: app/src/main/java/com/warchaser/musicplayer/tools/MusicInfo.kt
package com.warchaser.musicplayer.tools

import android.net.Uri
import android.os.Parcel
import android.os.Parcelable

/**
 * MusicInfo
 * bean
 * 传输数据类
 * */
class MusicInfo() : Parcelable{

    var id : Long = 0
    var title : String? = ""
    var album : String? = ""
    var albumId : Long = 0
    var duration : Int = 0
    var size : Long = 0
    var artist : String? = ""
    var url : String? = ""
    var pinyinInitial : String? = ""
    var pinyinTitle : String? = ""
    var uriWithCoverPic : String? = ""

    constructor(id: Long, title: String) : this(){
        this.id = id
        this.title = title
    }

    constructor(parcel: Parcel) : this() {
        id = parcel.readLong()
        title = parcel.readString()
        album = parcel.readString()
        albumId = parcel.readLong()
        duration = parcel.readInt()
        size = parcel.readLong()
        artist = parcel.readString()
        url = parcel.readString()
        pinyinInitial = parcel.readString()
        pinyinTitle = parcel.readString()
        uriWithCoverPic = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(id)
        parcel.writeString(title)
        parcel.writeString(album)
        parcel.writeLong(albumId)
        parcel.writeInt(duration)
        parcel.writeLong(size)
        parcel.writeString(artist)
        parcel.writeString(url)
        parcel.writeString(uriWithCoverPic)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MusicInfo> {
        override fun createFromParcel(parcel: Parcel): MusicInfo {
            return MusicInfo(parcel)
        }

        override fun newArray(size: Int): Array<MusicInfo?> {
            return arrayOfNulls<MusicInfo>(size)
        }
    }

    fun setUriWithCoverPic(uri: Uri) {
        this.uriWithCoverPic = uri.toString()
    }

}